packages/analytics-js-plugins/src/xhrQueue/utilities.ts

```diff
@@ -4,19 +4,19 @@ import {
   DATA_PLANE_API_VERSION,
   DEFAULT_RETRY_QUEUE_OPTIONS,
 } from './constants';
-import { mergeDeepRight, replaceNullValues } from '../utilities/common';
-import { QueueOpts, RudderEvent, RudderEventType, ILogger } from '../types/common';
+import { mergeDeepRight, stringifyWithoutCircular } from '../utilities/common';
+import { QueueOpts, RudderEvent, RudderEventType, ILogger, Nullable } from '../types/common';
 
 /**
  * Utility to get the stringified event payload
  * @param event RudderEvent object
  * @param logger Logger instance
  * @returns stringified event payload. Empty string if error occurs.
  */
-const getDeliveryPayload = (event: RudderEvent, logger?: ILogger): string => {
-  let deliveryPayloadStr = '';
+const getDeliveryPayload = (event: RudderEvent, logger?: ILogger): Nullable<string> | undefined => {
```

Comment:
fixed types.

stringifyWithoutCircular can return underined if undefined values is passed as argument

Reply:
Makes sense. In that case, let's make the return type `string | undefined` instead of type casting to `Nullable<string>`.
I originally forgot about the return type of the stringify function.

Reply:
if we know RudderEvent will never be undefined, based on argument type it will not be, we can leave the typecasting as it is the expected type for the usage at this point